### ISSUE TITLE
Add support for GitLab

### ISSUE THREAD

**Issue by To1ne** [gitgitgadget/gitgitgadget#1752]
Currently GitGitGadget works with GitHub. It would be nice to be able to have the same workflow using GitLab Merge Requests.

**Comment by dscho** [comment]
I agree! An important stepping stone for that is, ironically, the move away from Azure Pipelines to GitHub workflows, as that forces us away from persistent, self-hosted runners toward using ephemeral, hosted runners. A large part of that is tracked in #609, and a first step is #1473. You will also note that I refrained from using e.g. GitHub's [repository variables](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/store-information-in-variables#creating-configuration-variables-for-a-repository) to store the state, specifically to reduce the dependency on GitHub.
